### PR TITLE
GODRIVER-2072 Add race detector Evergreen variant.

### PR DESCRIPTION
**Pull request by benjirewis**
GODRIVER-2072

## Summary

Adds a new build variant to run tests in all packages with the [`-race`](https://go.dev/blog/race-detector) flag enabled. 

## Background & Motivation

The Go docs say:

>Because of its design, the race detector can detect race conditions only when they are actually triggered by running code, which means it’s important to run race-enabled binaries under realistic workloads. However, race-enabled binaries can use ten times the CPU and memory, so it is impractical to enable the race detector all the time. One way out of this dilemma is to run some tests with the race detector enabled. Load tests and integration tests are good candidates, since they tend to exercise concurrent parts of the code. Another approach using production workloads is to deploy a single race-enabled instance within a pool of running servers.

I _could_ add `-race` to every `go test` call in the Makefile, but I think we should run tests with `-race` in a separate task due to their nearly doubled test time. I've also made the assumption that OS, topology and server version are not particularly relevant and am only testing against an Ubuntu 18.04 replica set with the latest server version.

### CODE REVIEW

File: internal/uuid/uuid_test.go [hunk 1]
```diff
@@ -3,6 +3,13 @@
 // Licensed under the Apache License, Version 2.0 (the "License"); you may
 // not use this file except in compliance with the License. You may obtain
 // a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Turn off building when race detector is enabled. These tests cause race
+// detector errors pre go 1.19, as the race detector has a limit of 8128
+// goroutines.
+//
+//go:build !race
+// +build !race
```
Comment (matthewdale) [inline internal/uuid/uuid_test.go:12]: Can we use the `israce.Enabled` variable to skip individual tests instead of build tags?

**Review by matthewdale — COMMENTED** [review]
Adding tests with the race detector is really exciting! A few questions/suggestions.

File: Makefile [hunk 2]
```diff
@@ -116,7 +116,7 @@ build-aws-ecs-test:
 .PHONY: evg-test
 evg-test:
 	for TEST in $(TEST_PKGS); do \
-		go test -exec "env PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) LD_LIBRARY_PATH=$(LD_LIBRARY_PATH)" $(BUILD_TAGS) -v -timeout $(TEST_TIMEOUT)s $$TEST >> test.suite ; \
+		go test -exec "env PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) LD_LIBRARY_PATH=$(LD_LIBRARY_PATH)" $(BUILD_TAGS) -v -timeout $(TEST_TIMEOUT)s $(RACE) $$TEST >> test.suite ; \
```
[the remark this answers is not in the document]
Comment (benjirewis) [reply Makefile]: Yep! Good idea.

File: internal/uuid/uuid_test.go (hunk 1 above)
Comment (benjirewis) [reply internal/uuid/uuid_test.go]: Yep! Didn't even realize we had that 😮‍💨 .

**Review by matthewdale — APPROVED** [review]
Sorry for the slow review! Looks good 👍

**Pull request merged**